MathLib/LinAlg/RowColumnIndices.h

```diff
@@ -15,6 +15,21 @@
 namespace MathLib
 {
 
+#ifdef USE_PETSC
```

Comment:
The `#ifdef` is easily avoidable when using `std::vector<IDX_TYPE>*` with default `nullptr` value.